Libs/vtkAddon/vtkStreamingVolumeCodec.h

```diff
@@ -0,0 +1,146 @@
+/*==============================================================================
+
+Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+See COPYRIGHT.txt
+or http://www.slicer.org/copyright/copyright.txt for details.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+and was supported through CANARIE�s Research Software Program, and Cancer
+Care Ontario.
+
+==============================================================================*/
+
+#ifndef __vtkStreamingVolumeCodec_h
+#define __vtkStreamingVolumeCodec_h
+
+// vtkAddon includes
+#include "vtkAddon.h"
+#include "vtkStreamingVolumeFrame.h"
+
+// VTK includes
+#include <vtkImageData.h>
+#include <vtkObject.h>
+#include <vtkUnsignedCharArray.h>
+
+// STD includes
+#include <map>
+
+#ifndef vtkCodecNewMacro
+#define vtkCodecNewMacro(newClass) \
+vtkStandardNewMacro(newClass); \
+vtkStreamingVolumeCodec* newClass::CreateCodecInstance() \
+{ \
+return newClass::New(); \
+}
+#endif
+
+/// \brief VTK object for representing a volume compression codec (normally a video compression codec)
+/// Five functions from this class need to be implemented in child classes:
+/// 1. virtual bool EncodeImageDataInternal(vtkImageData* inputImageData, vtkStreamingVolumeFrame* outputFrame, bool forceKeyFrame);
+/// 2. virtual bool DecodeFrameInternal(vtkStreamingVolumeFrame* inputFrame, vtkImageData* outputImageData, bool saveDecodedImage);
+/// 3. virtual std::string UpdateParameterInternal(std::string parameterName, std::string parameterValue);
+/// 4. virtual std::string GetFourCC();
+/// 5. virtual std::string CreateCodecInstance(); // This can be overridden using vtkCodecNewMacro(className);
+class VTK_ADDON_EXPORT vtkStreamingVolumeCodec : public vtkObject
```

Comment:
We should add a Raw or Trivial codec that just copies the data. It would serve as an example, it could be also used for testing, and for allowing reading/writing uncompressed data with the same API as compressed data.